File: javascript/json-transform/src/functions/TransformerFunctionBoolean.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";
import { isTruthy } from "../JsonHelpers";

class TransformerFunctionBoolean extends TransformerFunction {
  constructor() {
    super({
      argsSets: [[{ name: "style", type: ArgType.String, defaultValue: "JAVA" }]],
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const jsStyle = (await context.getEnum("style")) === "JS";
    return isTruthy(await context.getUnwrapped(null), jsStyle);
  }
}

export default TransformerFunctionBoolean;

File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionBoolean.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.*;
import ai.bizone.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionBoolean extends TransformerFunction {
    public TransformerFunctionBoolean() {
        super(FunctionDescription.of(
            Map.of(
            "style", ArgumentType.of(ArgType.String).position(0).defaultValue("JAVA")
            )
        ));
    }

    @Override
    public Object apply(FunctionContext context) {
        var jsStyle = "JS".equals(context.getEnum("style"));
        var adapter = context.getAdapter();
        return adapter.isTruthy(context.getUnwrapped(null), jsStyle);
    }
}
